webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/TableColumnHeader/TableColumnHeader.tsx

```diff
@@ -65,6 +65,12 @@ export const TableColumnHeader = observer<RenderHeaderCellProps<any>>(function T
     dataGridContext.focus();
   }
 
+  useMemo(() => {
+    if (calculatedColumn.columnDataIndex) {
+      calculatedColumn.onRenderHeader?.(calculatedColumn.columnDataIndex);
+    }
+  }, [calculatedColumn, calculatedColumn.columnDataIndex]);
```

Comment:
no need for `calculatedColumn.columnDataIndex` dep

Reply:
thanks removed